src/wp-includes/meta.php

```diff
@@ -1399,6 +1399,10 @@ function sanitize_meta( $meta_key, $meta_value, $object_type, $object_subtype =
 function register_meta( $object_type, $meta_key, $args, $deprecated = null ) {
 	global $wp_meta_keys;
 
+	if ( isset( $deprecated ) ) {
+		_deprecated_argument( __FUNCTION__, '4.6.0' );
+	}
```

Comment:
This one is causing three test failures. These tests should be updated, but **not** to expect a deprecation notice (as it would stop the test at the deprecation and not actually run the test), but instead they should probably get a custom error handler to catch the deprecation and then ignore it.

A bit like this, but different: https://github.com/WordPress/Requests/pull/817/commits/4f3550a7287394ed9b00399d5d2904bd17a5cce6

Alternatively, the deprecation notice for `register_meta()` may need to be removed and annotated in the docblock why no call to `_deprecated_argument()` was added.

```
1) Tests_Meta_Register_Meta::test_register_meta_back_compat_with_auth_callback_and_no_sanitize_callback_has_old_style_auth_filter
Unexpected deprecation notice for register_meta.
Function register_meta was called with an argument that is deprecated since version 4.6.0 with no alternative available.
Failed asserting that an array is empty.

/var/www/tests/phpunit/includes/abstract-testcase.php:634
/var/www/tests/phpunit/includes/abstract-testcase.php:646
/var/www/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitGte8.php:80
/var/www/vendor/bin/phpunit:122

2) Tests_Meta_Register_Meta::test_register_meta_back_compat_with_auth_and_sanitize_callback_has_old_style_filters
Unexpected deprecation notice for register_meta.
Function register_meta was called with an argument that is deprecated since version 4.6.0 with no alternative available.
Failed asserting that an array is empty.

/var/www/tests/phpunit/includes/abstract-testcase.php:634
/var/www/tests/phpunit/includes/abstract-testcase.php:646
/var/www/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitGte8.php:80
/var/www/vendor/bin/phpunit:122

3) Tests_User_Capabilities::test_post_meta_caps
Unexpected deprecation notice for register_meta.
Function register_meta was called with an argument that is deprecated since version 4.6.0 with no alternative available.
Failed asserting that an array is empty.

/var/www/tests/phpunit/includes/abstract-testcase.php:634
/var/www/tests/phpunit/includes/abstract-testcase.php:646
/var/www/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitGte8.php:80
/var/www/vendor/bin/phpunit:122
```